### ISSUE TITLE
way to fail if image needs a disk

### ISSUE THREAD

**Issue by justincormack** [linuxkit/linuxkit#1755]
Docker (and some other containers) have a pretty poor failure mode if they don't get a persistent disk mount. We should have some way of failing hard if a disk is not mounted.

Possibly if any of the `onboot` containers fails we should just give up? Currently we do not check their exist codes. That might be too extreme, but maybe we should log something, it may be sensible though.

**Comment by eyz** [comment]
Yes, I thought about this issue a bit earloee, and agree that it is far too easy to miss a service failing startup. Perhaps a new flag to onboot containers to specify what the system should do on failure of that service.
